scw inspect: panic: runtime error: slice bounds out of range

```console
$ scw _flush-cache
Cache flushed
$ scw inspect alpine
- e4f44825          image:alpine-linux                                    Alpine Linux                                 arm
- ab8fbe9e          image:alpine-linux                                    Alpine Linux                                 x86_64
- 2a972c9e          image:x86-64-alpine-latest-2016-03-14-18-48           x86_64-alpine-latest-2016-03-14_18:48        x86_64
- 6b454797          snapshot:x86-64-alpine-latest-2016-03-14-18-48        x86_64-alpine-latest-2016-03-14_18:48        n/a
- 5bb7ebd2          snapshot:ocs-distrib-alpine-latest-2015-04-02-11-06   ocs-distrib-alpine-latest-2015-04-02_11:06   n/a
- 9e9a8d90          snapshot:scw-distrib-alpine-latest-2015-06-19-17-09   scw-distrib-alpine-latest-2015-06-19_17:09   n/a
- 4a1e1893          snapshot:ocs-distrib-alpine-latest-2015-04-02-11-50   ocs-distrib-alpine-latest-2015-04-02_11:50   n/a
- 8cb6255f          snapshot:ocs-distrib-alpine-latest-2015-04-02-09-39   ocs-distrib-alpine-latest-2015-04-02_09:39   n/a
panic: runtime error: slice bounds out of range

goroutine 22 [running]:
panic(0x4f3300, 0xc820012120)
	/usr/local/Cellar/go/1.6.1/libexec/src/runtime/panic.go:464 +0x3e6
github.com/scaleway/scaleway-cli/pkg/api.showResolverResults(0x7fff5fbff85c, 0x6, 0xc820444a00, 0x9, 0x10, 0x0, 0x0)
	/Users/moul/go/src/github.com/scaleway/scaleway-cli/pkg/api/api.go:1993 +0x7df
github.com/scaleway/scaleway-cli/pkg/api.InspectIdentifiers(0xc8200a2f00, 0xc82006e540, 0xc82006e5a0, 0x58c108, 0x1)
	/Users/moul/go/src/github.com/scaleway/scaleway-cli/pkg/api/helpers.go:233 +0x313
created by github.com/scaleway/scaleway-cli/pkg/commands.RunInspect
	/Users/moul/go/src/github.com/scaleway/scaleway-cli/pkg/commands/inspect.go:31 +0x142
```

--- a/README.md
+++ b/README.md
@@ -1184,6 +1184,7 @@ $ scw inspect myserver | jq '.[0].public_ip.address'
 
 ### master (unreleased)
 
+* `scw inspect` fix panic ([#353](https://github.com/scaleway/scaleway-cli/issues/353)
 * Clear cache between the releases ([#329](https://github.com/scaleway/scaleway-cli/issues/329)
 * Fix `scw _patch bootscript` nil dereference
 * Fix `scw images` bad error message ([#336](https://github.com/scaleway/scaleway-cli/issues/337))

--- a/pkg/api/api.go
+++ b/pkg/api/api.go
@@ -1981,16 +1981,14 @@ func (s *ScalewayAPI) GetServerID(needle string) (string, error) {
 }
 
 func showResolverResults(needle string, results ScalewayResolverResults) error {
-	arch := ""
 	w := tabwriter.NewWriter(os.Stderr, 20, 1, 3, ' ', 0)
 	defer w.Flush()
 	sort.Sort(results)
 	for _, result := range results {
-		arch = result.Arch
-		if arch == "" {
-			arch = "n/a"
+		if result.Arch == "" {
+			result.Arch = "n/a"
 		}
-		fmt.Fprintf(w, "- %s\t%s\t%s\t%s\n", result.TruncIdentifier(), result.CodeName(), result.Name, arch)
+		fmt.Fprintf(w, "- %s\t%s\t%s\t%s\n", result.TruncIdentifier(), result.CodeName(), result.Name, result.Arch)
 	}
 	return fmt.Errorf("Too many candidates for %s (%d)", needle, len(results))
 }

--- a/pkg/api/cache.go
+++ b/pkg/api/cache.go
@@ -8,6 +8,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"log"
 	"os"
 	"path/filepath"
 	"regexp"
@@ -93,6 +94,9 @@ type ScalewayResolverResults []ScalewayResolverResult
 
 // NewScalewayResolverResult returns a new ScalewayResolverResult
 func NewScalewayResolverResult(Identifier, Name, Arch string, Type int) ScalewayResolverResult {
+	if err := anonuuid.IsUUID(Identifier); err != nil {
+		log.Fatal(err)
+	}
 	return ScalewayResolverResult{
 		Identifier: Identifier,
 		Type:       Type,

--- a/pkg/api/cache.go
+++ b/pkg/api/cache.go
@@ -356,12 +356,12 @@ func (c *ScalewayCache) LookUpVolumes(needle string, acceptUUID bool) ScalewayRe
 	nameRegex := regexp.MustCompile(`(?i)` + regexp.MustCompile(`[_-]`).ReplaceAllString(needle, ".*"))
 	for identifier, fields := range c.Volumes {
 		if fields[CacheTitle] == needle {
-			entry := NewScalewayResolverResult(needle, fields[CacheTitle], fields[CacheArch], IdentifierVolume)
+			entry := NewScalewayResolverResult(identifier, fields[CacheTitle], fields[CacheArch], IdentifierVolume)
 			entry.ComputeRankMatch(needle)
 			exactMatches = append(exactMatches, entry)
 		}
 		if strings.HasPrefix(identifier, needle) || nameRegex.MatchString(fields[CacheTitle]) {
-			entry := NewScalewayResolverResult(needle, fields[CacheTitle], fields[CacheArch], IdentifierVolume)
+			entry := NewScalewayResolverResult(identifier, fields[CacheTitle], fields[CacheArch], IdentifierVolume)
 			entry.ComputeRankMatch(needle)
 			res = append(res, entry)
 		}

--- a/pkg/api/helpers.go
+++ b/pkg/api/helpers.go
@@ -236,45 +236,25 @@ func InspectIdentifiers(api *ScalewayAPI, ci chan ScalewayResolvedIdentifier, cj
 			ident := idents.Identifiers[0]
 			wg.Add(1)
 			go func() {
-				if ident.Type == IdentifierServer {
-					server, err := api.GetServer(ident.Identifier)
-					if err == nil {
-						cj <- InspectIdentifierResult{
-							Type:   ident.Type,
-							Object: server,
-						}
-					}
-				} else if ident.Type == IdentifierImage {
-					image, err := api.GetImage(ident.Identifier)
-					if err == nil {
-						cj <- InspectIdentifierResult{
-							Type:   ident.Type,
-							Object: image,
-						}
-					}
-				} else if ident.Type == IdentifierSnapshot {
-					snap, err := api.GetSnapshot(ident.Identifier)
-					if err == nil {
-						cj <- InspectIdentifierResult{
-							Type:   ident.Type,
-							Object: snap,
-						}
-					}
-				} else if ident.Type == IdentifierVolume {
-					volume, err := api.GetVolume(ident.Identifier)
-					if err == nil {
-						cj <- InspectIdentifierResult{
-							Type:   ident.Type,
-							Object: volume,
-						}
-					}
-				} else if ident.Type == IdentifierBootscript {
-					bootscript, err := api.GetBootscript(ident.Identifier)
-					if err == nil {
-						cj <- InspectIdentifierResult{
-							Type:   ident.Type,
-							Object: bootscript,
-						}
+				var obj interface{}
+				var err error
+
+				switch ident.Type {
+				case IdentifierServer:
+					obj, err = api.GetServer(ident.Identifier)
+				case IdentifierImage:
+					obj, err = api.GetImage(ident.Identifier)
+				case IdentifierSnapshot:
+					obj, err = api.GetSnapshot(ident.Identifier)
+				case IdentifierVolume:
+					obj, err = api.GetVolume(ident.Identifier)
+				case IdentifierBootscript:
+					obj, err = api.GetBootscript(ident.Identifier)
+				}
+				if err == nil && obj != nil {
+					cj <- InspectIdentifierResult{
+						Type:   ident.Type,
+						Object: obj,
 					}
 				}
 				wg.Done()

--- a/pkg/commands/inspect.go
+++ b/pkg/commands/inspect.go
@@ -101,7 +101,9 @@ func RunInspect(ctx CommandContext, args InspectArgs) error {
 			if ctx.Getenv("SCW_SENSITIVE") != "1" {
 				res = ctx.API.HideAPICredentials(res)
 			}
-			fmt.Fprintln(ctx.Stdout, res)
+			if len(res) > 2 {
+				fmt.Fprintln(ctx.Stdout, res)
+			}
 		}
 	}
 

